pkg/kv/kvserver/kvserverpb/proposer_kv.proto

```diff
@@ -258,6 +252,22 @@ message ReplicatedEvalResult {
   // application while ensuring correctness in the case where the lease transfer
   // is applied on the new leaseholder through a Raft snapshot.
   kv.kvserver.readsummary.ReadSummary prior_read_summary = 22;
+  
+  // LinkExternalSSTable, like AddSSTable, is a side effect that must execute
+  // before the Raft application is committed. It must be idempotent to account
+  // for an ill-timed crash after applying the side effect, but before
+  // committing the batch.
+  message LinkExternalSSTable {
```

Comment:
this struct does differ from the request one, e.g. mvcc stats need to make it to kv server but don't go into (this part) of the proposal so I think this justifies the somewhat duplicated structs.

Reply:
i.e. we shouldn't pass the `ExternalFile` message currently defined in api.proto? agreed.